chart/compass/templates/registration-job.yaml

```diff
@@ -141,12 +142,71 @@ spec:
                 CLIENT_SECRET="${CLIENT_SECRET:1:${#CLIENT_SECRET}-2}"
                 TOKENS_ENDPOINT="${TOKENS_ENDPOINT:1:${#TOKENS_ENDPOINT}-2}"
 
-                kubectl create secret generic ${SECRET_NAME} --from-literal=client_id=$CLIENT_ID --from-literal=client_secret=$CLIENT_SECRET --from-literal=tokens_endpoint=$TOKENS_ENDPOINT -n {{ .Release.Namespace }}
+                kubectl create secret generic ${PROVISIONER_SECRET_NAME} --from-literal=client_id=$CLIENT_ID --from-literal=client_secret=$CLIENT_SECRET --from-literal=tokens_endpoint=$TOKENS_ENDPOINT -n {{ .Release.Namespace }}
 
                 echo "Compass Provisioner registration as integration system and requesting credentials - finished."
                 set -o pipefail
               }
 
+              function register_broker() {
+                set +o pipefail
+                echo "Kyma environment broker registration as integration system - in progress."
+
+                RESPONSE_BODY=$(curl -k "${DIRECTOR_URL}" \
+                  -H 'Content-Type: application/json' \
+                  -H "authorization: Bearer ${INTERNAL_TOKEN}" \
+                  --data-binary '{
+                    "query":"mutation registerIntegrationSystem {\n  result: registerIntegrationSystem(\n    in: { name: \"compass-kyma-broker\", description: \"Compass Kyma Environment Broker\" }\n  ) {\n    id\n  }\n}\n"
+                  }')
+                echo $RESPONSE_BODY
+
+                INTEGRATION_SYSTEM_ID=$(echo $RESPONSE_BODY | jq -e '.data .result .id')
+                INTEGRATION_SYSTEM_ID="${INTEGRATION_SYSTEM_ID:1:${#INTEGRATION_SYSTEM_ID}-2}"
+
+                CREDENTIAL_RESPONSE_BODY=$(curl -k "${DIRECTOR_URL}" \
+                  -H 'Content-Type: application/json' \
+                  -H 'Accept: application/json' \
+                  -H 'Connection: keep-alive' \
+                  -H "authorization: Bearer ${INTERNAL_TOKEN}" \
+                  --data-binary '{"query":"mutation requestCredentials {\n\trequestClientCredentialsForIntegrationSystem (id : \"'"$INTEGRATION_SYSTEM_ID"'\") {\n    id\n    auth {\n      credential {\n        ... on BasicCredentialData {\n            username\n            password\n        }\n        ...  on OAuthCredentialData {\n            clientId\n            clientSecret\n            url\n        }\n      }\n    }\n  }\n}\n"
+                 }')
+
+                CLIENT_ID=$(echo $CREDENTIAL_RESPONSE_BODY | jq -e '.data .requestClientCredentialsForIntegrationSystem .auth .credential .clientId')
+                CLIENT_SECRET=$(echo $CREDENTIAL_RESPONSE_BODY | jq -e '.data .requestClientCredentialsForIntegrationSystem .auth .credential .clientSecret')
+                TOKENS_ENDPOINT=$(echo $CREDENTIAL_RESPONSE_BODY | jq -e '.data .requestClientCredentialsForIntegrationSystem .auth .credential .url')
```

Comment:
We duplicate a lot of code here. Maybe we could create a function `registerIntegrationSystem(name, description, secretName)` and just use it twice, once for Provisioner and once for Broker?

Reply:
You are right, I created shared methods:
`ensure_secret(name, description, secretName)`
`register_integration_system(name, description, secretName)`

and they are executed like that:
```
              echo "Checking if provisioner secret with credentials exists..."
              ensure_secret "compass-provisioner" "Compass Provisioner" $PROVISIONER_SECRET_NAME

              echo "Checking if broker secret with credentials exists..."
              ensure_secret "compass-kyma-environment-broker" "Compass Kyma Environment Broker" $BROKER_SECRET_NAME
```